From ba187498999bb4c4d7fa9249a16f0a42e0ed7989 Mon Sep 17 00:00:00 2001
From: khoahvd <khoahvd@fsoft.com.vn>
Date: Mon, 28 Dec 2015 14:49:46 +0700
Subject: [PATCH 6/8] Flexible_Display_Support_H264StreamingSession_h

Signed-off-by: khoahvd <khoahvd@fsoft.com.vn>
---
 .../H264Streaming/qgstreamerH264Streamingsession.h | 41 +++++++++++++++-------
 1 file changed, 29 insertions(+), 12 deletions(-)

diff --git a/src/plugins/gstreamer/H264Streaming/qgstreamerH264Streamingsession.h b/src/plugins/gstreamer/H264Streaming/qgstreamerH264Streamingsession.h
index b7219dd..f4e95da 100644
--- a/src/plugins/gstreamer/H264Streaming/qgstreamerH264Streamingsession.h
+++ b/src/plugins/gstreamer/H264Streaming/qgstreamerH264Streamingsession.h
@@ -73,6 +73,21 @@ typedef enum {
   GST_AUTOPLUG_SELECT_SKIP
 } GstAutoplugSelectResult;
 
+typedef struct user_data_t{
+    GstElement *m_playbin;
+    GstElement *decoder, *videosink, *parser;
+    GstPad *blockpad, *tee_q1_pad;
+    bool m_wls_qos;
+    int m_wls_position_x;
+    int m_wls_position_y;
+    int m_wls_out_width;
+    int m_wls_out_height;
+    bool m_transmit;
+    int m_vspm_width;
+    int m_vspm_height;
+    int counter;
+}user_data_t;
+
 class QGstreamerH264StreamingSession : public QObject,
                                 public QGstreamerBusMessageFilter
 {
@@ -107,6 +122,9 @@ public:
     void setWlsOutHeight(int wlsoutheight);
     void componentComplete();
 
+    static GstPadProbeReturn pad_probe_cb (GstPad * pad, GstPadProbeInfo * info, gpointer user_data);
+    static GstPadProbeReturn event_probe_cb (GstPad * pad, GstPadProbeInfo * info, gpointer user_data);
+
     GstElement *playbin() const;
     QGstreamerBusHelper *bus() const { return m_busHelper; }
 
@@ -236,12 +254,12 @@ private:
     QH264Streaming::State m_videostate;
     QH264Streaming::State m_pendingState;
     QGstreamerBusHelper* m_busHelper;
-    GstElement *m_playbin;
+
     
     QBasicTimer timer;
     GstPadTemplate *tee_src_pad_template;
-    GstPad *tee_q1_pad, *tee_q2_pad;
-	GstPad *q1_pad, *q2_pad;
+    GstPad  *tee_q2_pad, *tee_q3_pad;
+    GstPad *q1_pad, *q2_pad, *q3_pad;
 
     GstElement* m_videoOutputBin;
     GstElement* m_videoIdentity;
@@ -304,7 +322,7 @@ private:
     gulong pad_probe_id;
 
     GstCaps *filtercaps_vspm, *filtercaps_vspm1, *filtercaps_depay, *filtercaps_parse, *streaming_src_cap, *filtercaps_dec;
-    GstElement *udp_src, *udp_sink, *pay, *depay, *parser, *encoder, *decoder, *filter, *capfilter_vspm, *videoIdentity_depay, *videoIdentity_parse, *videosink, *tee, *filter1, *capfilter_vspm1, *parser1;
+    GstElement *udp_src, *udp_sink, *pay, *depay, *decoder1, *encoder, *videoIdentity_depay, *videoIdentity_parse, *tee, *filter1, *capfilter_vspm1, *parser1, *queue, *fsink;
     GstStateChangeReturn ret;
     GstState currentState, newState;
     GstState pending;
@@ -316,24 +334,23 @@ private:
     bool m_no_reorder;
     QString m_udp_port;
     QString m_udp_host;
-    bool m_transmit;
+
     bool m_vspm;
     bool m_vspm_dmabuf_use;
-    int m_vspm_width;
+
     int m_vspm_width1;
-    int m_vspm_height;
+
     int m_vspm_height1;
     QString m_vspm_format;
     qint64 m_wls_max_lateness;
-    bool m_wls_qos;
-    int m_wls_position_x;
-    int m_wls_position_y;
-    int m_wls_out_width;
-    int m_wls_out_height;
+
     int count;
     bool notFirstRun;
     bool isFirstRemove;
     bool isTeePadAdd;
+    bool isFirstChange;
+
+    user_data_t userData;
 };
 
 QT_END_NAMESPACE
-- 
1.9.1

