From fc5809e0fd7e8c95cf96415cabf8ca6052fef54d Mon Sep 17 00:00:00 2001
From: TraHT <traht@fsoft.com.vn>
Date: Thu, 28 Jun 2018 10:50:15 +0700
Subject: Modify 1004-window.c-do-not-resize-if-not-needed.patch to fix hunk

Signed-off-by: TraHT <traht@fsoft.com.vn>
---
 clients/window.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/clients/window.c b/clients/window.c
index 95796d4..a1a3fe2 100644
--- a/clients/window.c
+++ b/clients/window.c
@@ -555,6 +555,7 @@ struct egl_window_surface {
 	struct wl_surface *surface;
 	struct wl_egl_window *egl_window;
 	EGLSurface egl_surface;
+	int w, h;
 };
 
 static struct egl_window_surface *
@@ -572,8 +573,12 @@ egl_window_surface_prepare(struct toysurface *base, int dx, int dy,
 
 	surface_to_buffer_size (buffer_transform, buffer_scale, &width, &height);
 
-	wl_egl_window_resize(surface->egl_window, width, height, dx, dy);
-	cairo_gl_surface_set_size(surface->cairo_surface, width, height);
+	if (width != surface->w || height != surface->h) {
+		wl_egl_window_resize(surface->egl_window, width, height, dx, dy);
+		cairo_gl_surface_set_size(surface->cairo_surface, width, height);
+		surface->w = width;
+		surface->h = height;
+	}
 
 	return cairo_surface_reference(surface->cairo_surface);
 }
@@ -676,6 +681,9 @@ egl_window_surface_create(struct display *display,
 	surface->display = display;
 	surface->surface = wl_surface;
 
+	surface->w = rectangle->width;
+	surface->h = rectangle->height;
+
 	surface->egl_window = wl_egl_window_create(surface->surface,
 						   rectangle->width,
 						   rectangle->height);
-- 
2.7.4

