From 8f8615820b3edf435e565b85e387c25fbdf7f9f1 Mon Sep 17 00:00:00 2001
From: TraHT <traht@fsoft.com.vn>
Date: Thu, 28 Jun 2018 10:53:02 +0700
Subject: Modify 1007-set-position-to-fix-problem-app-under-weston-bar.patch to
 fix hunk

Signed-off-by: TraHT <traht@fsoft.com.vn>
---
 desktop-shell/shell.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index f80088f..3cfe95f 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -4084,6 +4084,7 @@ weston_view_set_initial_position(struct weston_view *view,
 	int ix = 0, iy = 0;
 	int32_t range_x, range_y;
 	int32_t x, y;
+	int32_t panel_width = 0, panel_height = 0;
 	struct weston_output *output, *target_output = NULL;
 	struct weston_seat *seat;
 	pixman_rectangle32_t area;
@@ -4120,6 +4121,7 @@ weston_view_set_initial_position(struct weston_view *view,
 	 * If this is negative it means that the surface is bigger than
 	 * output.
 	 */
+	get_output_panel_size(shell, output, &panel_width, &panel_height);
 	get_output_work_area(shell, target_output, &area);
 
 	x = area.x;
@@ -4133,6 +4135,9 @@ weston_view_set_initial_position(struct weston_view *view,
 	if (range_y > 0)
 		y += random() % range_y;
 
+	if(y < panel_height){
+       y = y + panel_height;
+    }
 	weston_view_set_position(view, x, y);
 }
 
-- 
2.7.4

