From c80bcc8824a39aa48052623926ce61fdc790da14 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Thu, 8 Sep 2016 19:26:18 +0700
Subject: [PATCH 2/2] update-weston-bar-larger-for-touch-easier

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 clients/desktop-shell.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clients/desktop-shell.c b/clients/desktop-shell.c
index 9cf3d68..e3029de 100644
--- a/clients/desktop-shell.c
+++ b/clients/desktop-shell.c
@@ -469,7 +469,7 @@ panel_configure(void *data,
	struct surface *surface = window_get_user_data(window);
	struct panel *panel = container_of(surface, struct panel, base);

-	window_schedule_resize(panel->window, width, 32);
+	window_schedule_resize(panel->window, width, 50);
 }

 static void
--
1.9.1
