From 1da99daf4b8aa494453f59dac7b1802e4c815c01 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Wed, 16 Nov 2016 11:05:33 +0700
Subject: [PATCH] Fix-issue-can-not-open-client-server-app-at-the-same

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 content/Networking_Model.qml | 14 +++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/content/Networking_Model.qml b/content/Networking_Model.qml
index 753cc42..49aff17 100644
--- a/content/Networking_Model.qml
+++ b/content/Networking_Model.qml
@@ -1,14 +1,8 @@
 import QtQuick 2.2
 
 ListModel {
-	ListElement{name: "Network Server"; 
-					path:"/usr/share/qt5/examples/network/fortuneserver/fortuneserver";
-					detail:"Network Communication\n\nfortuneserver: accept incoming TCP";
-					image: "images/networking_and_connectivity/remote-desktop.svg"}
-
-	ListElement{name: "Network Client"; 
-					path:"/usr/share/qt5/examples/network/fortuneclient/fortuneclient";
-					detail:"Network Communication\n\nfortunesclient: request a line of text from server";
-					image: "images/networking_and_connectivity/knetattach.svg"}
-	
+	ListElement{name: "Network demo";
+                    path:"/home/root/qt_framework/network_demo_script.sh";
+                    detail:"Network Communication\n\nfortuneclient-server: TCP communication";
+                    image: "images/networking_and_connectivity/remote-desktop.svg"}
 }
-- 
1.9.1

