From 5457894d33f59095a31e0263a54b13349ce732bb Mon Sep 17 00:00:00 2001
From: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
Date: Thu, 17 Aug 2017 09:42:16 +0700
Subject: [PATCH] Add fake button for create virtual keyboard for r8a7743

Signed-off-by: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7743-iwg20m.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7743-iwg20m.dts b/arch/arm/boot/dts/r8a7743-iwg20m.dts
index f7950f0..fb9d753 100644
--- a/arch/arm/boot/dts/r8a7743-iwg20m.dts
+++ b/arch/arm/boot/dts/r8a7743-iwg20m.dts
@@ -62,6 +62,18 @@
 		regulator-boot-on;
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		key-a {
+			gpios = <&gpio7 19 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_A>;
+			label = "SW30";
+			gpio-key,wakeup;
+			debounce-interval = <20>;
+		};
+	};
+
 	vcc_sdhi1: reg_vcc_sdhi1 {
 		compatible = "regulator-fixed";
 
-- 
1.9.1

