From fcc1b7cd2fabb696ba465e5db3170e9633c1d70a Mon Sep 17 00:00:00 2001
From: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
Date: Fri, 9 Mar 2018 13:49:06 +0700
Subject: [PATCH] Add-fake-button-for-create-virtual-keyboard-for-r8a7744

Signed-off-by: Quoc Ngo <quoc.ngo.wz@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7744-iwg20m.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744-iwg20m.dts b/arch/arm/boot/dts/r8a7744-iwg20m.dts
index 5bfafc6..ccaa8b0 100644
--- a/arch/arm/boot/dts/r8a7744-iwg20m.dts
+++ b/arch/arm/boot/dts/r8a7744-iwg20m.dts
@@ -12,6 +12,7 @@
 /dts-v1/;
 #include "r8a7744.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
 
 / {
 	model = "iwg20m";
@@ -55,6 +56,18 @@
 		regulator-boot-on;
 	};
 
+	gpio-keys {
+		compatible = "gpio-keys";
+
+		key-a {
+			gpios = <&gpio7 19 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_A>;
+			label = "SW30";
+			gpio-key,wakeup;
+			debounce-interval = <20>;
+		};
+	};
+
 	vcc_sdhi1: reg_vcc_sdhi1 {
 		compatible = "regulator-fixed";
 
-- 
1.9.1

