diff --git a/sys/v4l2/Makefile.am b/sys/v4l2/Makefile.am
index da138ae..3e16dcd 100644
--- a/sys/v4l2/Makefile.am
+++ b/sys/v4l2/Makefile.am
@@ -37,6 +37,7 @@ libgstvideo4linux2_la_LIBTOOLFLAGS = $(GST_PLUGIN_LIBTOOLFLAGS)
 libgstvideo4linux2_la_LIBADD =   $(GST_PLUGINS_BASE_LIBS) \
 				 -lgstallocators-$(GST_API_VERSION) \
 				 -lgstvideo-$(GST_API_VERSION) \
+				 -lmemcpy \
 				 $(GST_BASE_LIBS) \
 				 $(GST_LIBS) \
 				 $(xv_libs) \
diff --git a/sys/v4l2/gstv4l2src.c b/sys/v4l2/gstv4l2src.c
index b14766c..6288d60 100644
--- a/sys/v4l2/gstv4l2src.c
+++ b/sys/v4l2/gstv4l2src.c
@@ -55,7 +55,8 @@
 #include <gst/video/gstvideopool.h>
 
 #include "gstv4l2src.h"
-
+#include "memcpy_unalign.h"
+ 
 #include "gstv4l2colorbalance.h"
 #include "gstv4l2tuner.h"
 #ifdef HAVE_XVIDEO
@@ -899,7 +900,7 @@ gst_v4l2src_fill (GstPushSrc * src, GstBuffer * buf)
     ptr_tmp += vin_width * vin_height;
     
     for (i=1; i*2 < vin_height; i++)
-        memcpy((ptr_tmp + i*vin_width), (ptr_tmp + i*vin_width*2), vin_width);
+        memcpy_unalign((ptr_tmp + i*vin_width), (ptr_tmp + i*vin_width*2), vin_width);
 
     size_new = map.size *3/4;
     gst_buffer_unmap (buf, &map);
