From 46691aea2813894ca94c6f89d5d95ee4000d3974 Mon Sep 17 00:00:00 2001
From: nhannguyen <nhan.nguyen.yb@renesas.com>
Date: Wed, 29 Nov 2017 14:59:43 +0700
Subject: [PATCH] Remove IP show and terminal icon on HMI launcher

Signed-off-by: nhannguyen <nhan.nguyen.yb@renesas.com>
---
 weston.ini | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/weston.ini b/weston.ini
index 9fd15c7..40ec3db 100644
--- a/weston.ini
+++ b/weston.ini
@@ -33,9 +29,5 @@ icon=/home/root/launcher-taskbar/hmi-icons/Demo3D.png
 path=/home/root/launcher-taskbar/hmi-scripts/Demo3D.sh
 
 [launcher]
-icon=/home/root/launcher-taskbar/hmi-icons/IP-show.png
-path=/home/root/launcher-taskbar/hmi-scripts/ip_show
-
-[launcher]
 icon=/home/root/launcher-taskbar/hmi-icons/Exit.png
 path=/home/root/launcher-taskbar/hmi-scripts/Exit.sh
-- 
1.9.1

