diff --git a/omx/gstomxvideoenc.c b/omx/gstomxvideoenc.c
index c378608..c3a9ef2 100644
--- a/omx/gstomxvideoenc.c
+++ b/omx/gstomxvideoenc.c
@@ -1437,6 +1437,7 @@ gst_omx_video_enc_set_format (GstVideoEncoder * encoder,
         if (gst_omx_port_allocate_buffers (self->enc_in_port) != OMX_ErrorNone)
           return FALSE;
       }
+	  self->p_buffers = p_buffers;
     } else {
       if (gst_omx_port_allocate_buffers (self->enc_in_port) != OMX_ErrorNone)
         return FALSE;
