From a5945745a4c2a72ba964574a339236b098cd5467 Mon Sep 17 00:00:00 2001
From: "Nguyen Van linh [ FGA.BU13.RZG ]" <LinhNV25@fsoft.com.vn>
Date: Fri, 29 Jun 2018 15:11:45 +0700
Subject: [PATCH] Workaround issue video encode call set property 2 times using
  wrong value

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
Signed-off-by: Nguyen Van linh [ FGA.BU13.RZG ] <LinhNV25@fsoft.com.vn>
---
 omx/gstomxvideoenc.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/omx/gstomxvideoenc.c b/omx/gstomxvideoenc.c
index c54d8a7..a66e13a 100644
--- a/omx/gstomxvideoenc.c
+++ b/omx/gstomxvideoenc.c
@@ -413,11 +413,17 @@ static void
 gst_omx_video_enc_set_property (GObject * object, guint prop_id,
     const GValue * value, GParamSpec * pspec)
 {
+  /* FIX ME */
+  // Work around issue call func gst_omx_video_enc_set_property() 2 times lead to wrong values.
+  static gboolean repeat = 0;
   GstOMXVideoEnc *self = GST_OMX_VIDEO_ENC (object);
 
   switch (prop_id) {
     case PROP_CONTROL_RATE:
-      self->control_rate = g_value_get_enum (value);
+      if (!repeat){
+            repeat = 1;
+            self->control_rate = g_value_get_enum (value);
+      }
       break;
     case PROP_TARGET_BITRATE:
       self->target_bitrate = g_value_get_uint (value);
-- 
2.7.4

