From b42ce286ff7c1205f35835acabc74ecccc1a71cf Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Mon, 14 Nov 2016 16:26:06 +0700
Subject: [PATCH] Add-fake-button-for-create-virtual-keyboard-for-r8a7744

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 arch/arm/boot/dts/r8a7744-iwg20m.dts | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm/boot/dts/r8a7744-iwg20m.dts b/arch/arm/boot/dts/r8a7744-iwg20m.dts
index 5ca3207..64a2cd0 100644
--- a/arch/arm/boot/dts/r8a7744-iwg20m.dts
+++ b/arch/arm/boot/dts/r8a7744-iwg20m.dts
@@ -48,6 +48,18 @@
 		regulator-always-on;
 	};
 
+        gpio-keys {
+                compatible = "gpio-keys";
+
+                key-a {
+                        gpios = <&gpio7 19 GPIO_ACTIVE_LOW>;
+                        linux,code = <KEY_A>;
+                        label = "SW30";
+                        gpio-key,wakeup;
+                        debounce-interval = <20>;
+                };
+        };
+
 	vcc_sdhi1: reg_vcc_sdhi1 {
 		compatible = "regulator-fixed";
 
-- 
1.9.1

