From 86c3eaacafffd7174b75441db77f17f171a9b7eb Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Thu, 12 May 2016 13:41:08 +0700
Subject: [PATCH] Update flexible size Qt WebEngine

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>

diff --git a/examples/webengine/quicknanobrowser/BrowserWindow.qml b/examples/webengine/quicknanobrowser/BrowserWindow.qml
index 8a7dc4a..f67c692 100644
--- a/examples/webengine/quicknanobrowser/BrowserWindow.qml
+++ b/examples/webengine/quicknanobrowser/BrowserWindow.qml
@@ -54,8 +54,8 @@ ApplicationWindow {
     property Item currentWebView: tabs.currentIndex < tabs.count ? tabs.getTab(tabs.currentIndex).item : null
     property int previousVisibility: Window.Windowed

-    width: 1300
-    height: 900
+    width: screenWidth - 10	// -10 for easier to click exit button
+    height: screenHeight
     visible: true
     title: currentWebView && currentWebView.title

diff --git a/examples/webengine/quicknanobrowser/main.cpp b/examples/webengine/quicknanobrowser/main.cpp
index b7ab036..f2cccb9 100644
--- a/examples/webengine/quicknanobrowser/main.cpp
+++ b/examples/webengine/quicknanobrowser/main.cpp
@@ -50,6 +50,7 @@ typedef QGuiApplication Application;
 #include <QtQml/QQmlApplicationEngine>
 #include <QtQml/QQmlContext>
 #include <QtWebEngine/qtwebengineglobal.h>
+#include <QtGui/QScreen>

 static QUrl startupUrl()
 {
@@ -71,10 +72,17 @@ int main(int argc, char **argv)
     Application app(argc, argv);

     QtWebEngine::initialize();
-
-    QQmlApplicationEngine appEngine;
+
+	QQmlApplicationEngine appEngine;
     Utils utils;
     appEngine.rootContext()->setContextProperty("utils", &utils);
+	QScreen *screen = QGuiApplication::screens().at(1);
+    if (screen == NULL)
+		screen = QGuiApplication::screens().at(0);
+
+    appEngine.rootContext()->setContextProperty("screenWidth", screen->availableSize().width());
+    appEngine.rootContext()->setContextProperty("screenHeight", screen->availableSize().height());
+
     appEngine.load(QUrl("qrc:/ApplicationRoot.qml"));
     QMetaObject::invokeMethod(appEngine.rootObjects().first(), "load", Q_ARG(QVariant, startupUrl()));

--
1.9.1
