From f4cf4457cefd2afa30ceff93cc361f23e6817c9b Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Thu, 22 Sep 2016 20:34:53 +0700
Subject: [PATCH] Use less effect config for RZG1E to improve performance

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 main_lofi.qml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/main_lofi.qml b/main_lofi.qml
index 8ebd802..9af1b68 100644
--- a/main_lofi.qml
+++ b/main_lofi.qml
@@ -40,8 +40,8 @@

 DemoMain {

-    width: 854
-    height: 480
+    width: 1280
+    height: 720

     useDropShadow: false;
     useSimpleGradient: true;
--
1.9.1
